gtsam/hybrid/DCFactor.h

```diff
@@ -99,10 +111,24 @@ class DCFactor : public Factor {
    * @param discreteVals - Likewise, assignment to the discrete variables in
    * `discreteKeys__`.
    */
-  virtual boost::shared_ptr<GaussianFactor> linearize(
+  virtual GaussianFactor::shared_ptr linearize(
```

Comment:
revert and forward declare GaussianFactor, so we don't have to include GaussianFactor.h and co here.

Reply:
I don't quite understand this. GaussianFactor.h is not included here, so do you want to remove the #include from the DCGaussianMixtureFactor?

Reply:
If it is not included, it should not compile. If it is included, it should be removed from this file.

Reply:
Okay then there's nothing to do here since we include `GaussianFactorGraph.h` (which is needed) and that adds `GaussianFactor` as well. :)

Reply:
ok